x/x.go

```diff
@@ -619,7 +619,7 @@ func Max(a, b uint64) uint64 {
 func RetryUntilSuccess(maxRetries int, waitAfterFailure time.Duration,
 	f func() error) error {
 	var err error
-	for retry := maxRetries; retry != 0; retry-- {
+	for retry := maxRetries; retry > 0; retry-- {
```

Comment:
so this fails transactions with large keys ?

Reply:
No, Sorry, this issue doesn't fix the transaction with large keys. Currently if we have large keys, the write to badger api call fails. Once that request fails, the alpha gets stuck in a infinite loop, which causes it to not process anything further. This diff fixes the infinite loop. Aman has a diff, where we parse keys and fail the request before hand itself.